packages/web-react/src/hooks/useScrollControl.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react';
+
+const SCROLLING_DISABLED_CLASSNAME = 'is-scrolling-disabled';
+const DIALOG_OPEN_SELECTOR = 'dialog[open]';
+
+const disableScroll = () => {
+  const { body } = document;
+  if (body.classList.contains(SCROLLING_DISABLED_CLASSNAME)) {
+    return;
+  }
+
+  const scrollBarWidth = window.innerWidth - body.clientWidth;
+  const offsetY = window.scrollY;
+
+  body.style.paddingRight = `${scrollBarWidth}px`;
+  body.style.top = `-${offsetY}px`;
+  body.classList.add(SCROLLING_DISABLED_CLASSNAME);
+};
+
+const enableScroll = () => {
+  const dialogs = document.querySelectorAll(DIALOG_OPEN_SELECTOR);
```

Comment:
⚠️ Beware, you are now controlling all the dialogs on the page. I think, from the usage in the `useDialog` hook, that we should control only the one which is passed through the `ref` prop.

However, I think that the `ref` prop will always be set so the underlying condition may be useless because you are only appending styling to the body based on the `isOpen` state.

Reply:
Tru, modified to use ref. Thanks!